Add installation instructions

mt7479 opened the pull request:
I added some installation instructions and example files for less experienced users.

Maybe check config.example.py, I'm not sure if the SECRET_KEY variable needs to be specified with a b in front. However everything else threw tracebacks that a bytes object is expected.

horazont commented:
Thanks, much appreciated.

I have a few issues with this though:

1. It needs to be made very clear that this is one of many ways to set this up. I for myself would for example recommend to run this as a wsgi component, not via the flask http server for performance (since we’re sending files off the FS, using wsgi can yield a lot of improvement here).

2. I would prefer to recommend to install the dependencies (which is really just flask) via the distributions package manager; this will give us a consistent path to ``/usr/bin/flask`` across all platforms. Simply skip the setup.py step and point the service file to the original code clone.

What do you think?

mt7479 commented:
I changed the flask path in the systemd unit.

Unfortunately i didn't notice you had updated the readme, i've merged and added some more clarifications form my side in a single commit.

Running this with wsgi didn't even occur to me, if added some hints in the readme. You might want to fix this up yourself since i don't know the limitations/issues of wsgi.

mt7479 commented:
If you are going to write some bits about wsgi integration it might be better not to clutter the main readme with all this, but to move specific instructions to doc/wsgi.md doc/flask-proxy.md etc.

horazont commented on README.rst:
```diff
@@ -78,3 +82,6 @@ Configure your webserver:
 
 As final step you need to point your external webserver to your xmpp-http-upload flask app.
 Check the ``contrib`` directory, there is an example for nginx there.
+=======
```
That merge somehow went wrong. This line should be blank and the items belong to the list at the end of the previous section.

(Don’t worry if you don’t want to, I’ll clean this up when I merge the PR otherwise.)

mt7479 commented:
I'll keep trying 😄
